packages/es-components/src/components/controls/buttons/LinkButton.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import classnames from 'classnames';
+import { useTheme } from '../../util/useTheme';
+
+const StyledButton = styled.button`
+  background-color: transparent;
+  border: none;
+  box-shadow: none;
+  color: ${props => props.variant.bgColor};
+  font-size: inherit;
+  line-height: ${props => props.theme.sizes.baseLineHeight};
+  padding: 0;
+  text-decoration: underline;
+
+  &:hover,
+  &:focus,
+  &:active {
+    color: ${props => props.variant.hoverBgColor};
+    text-decoration: none;
+  }
+
+  &:active {
+    box-shadow: none;
+  }
+
+  &[disabled]:hover {
+    color: ${props => props.variant.bgColor};
+    text-decoration: underline;
+  }
+`;
+
+function InnerButton({
+  children,
+  styleType,
+  size,
+  block,
+  innerRef,
+  ...buttonProps
+}) {
+  const theme = useTheme();
+  const variant = theme.buttonStyles.button.variant[styleType];
+  const { className, ...otherProps } = buttonProps;
+  const sharedProps = {
+    block,
+    variant,
+    ref: innerRef,
+    ...otherProps
+  };
+
+  return (
+    <StyledButton
+      {...sharedProps}
+      className={classnames('es-button es-button--link', className)}
+    >
+      {children}
+    </StyledButton>
+  );
+}
+
+const LinkButton = React.forwardRef((props, ref) => (
+  <InnerButton innerRef={ref} {...props} />
+));
+
+LinkButton.propTypes = {
```

Comment:
A few notes on the `propTypes`:

Do `LinkButton` components have different styles based on the `styleType`? If not, it can be removed.

I don't think the `size` prop applies to `LinkButton`.  It can be removed if it's not used.

I don't think `LinkButton` should ever be allowed to be a block element. I think we can safely remove `block` as a prop for this component.

Reply:
Good catches -- I thought of a couple of these and then forgot to address them. I did leave the styleType intentionally because it does change the color of the text which I think could be useful. Size and block can definitely go.